### ISSUE TITLE
Infinite loop when activating/deactivating a package

### ISSUE THREAD

**Issue by ohnx** [haikuports/haikuporter#102]
Hi!

I'm trying to build [a certain recipe](https://github.com/ohnx/haikuports/blob/master/games-strategy/Haiku2048/Haiku2048.recipe), but it seems like it gets stuck in an infinite loop here:

https://github.com/haikuports/haikuporter/blob/5fc67389ac0cb7b34dfba1ada14aa501e01649c3/HaikuPorter/BuildPlatform.py#L193

Would it be possible to limit this in say a `for l in xrange(50)` so that it won't run forever?

**Comment by mmuman** [comment]
Yes it happens quite often here, either on startup or after a build...
I usually work around by touching or removing the file in question but it's painful.

**Comment by fbrosson** [comment]
I also get these infinite loops very often. I think I get these randomly.
I just hit Ctrl-C and retry ;-) Sometimes I have to retry several times before it really starts building.
Maybe I should try and see if touch has a positive impact.

**Comment by fbrosson** [comment]
I forgot to mention that sometimes the infinite loop happens **_after_** the package(s) is/are built.
So if haikuporter has been running for a while before it starts looping I always have a look in work-\*/hpkgs/ (right after hitting Ctrl-C) to see if the hpkg files were left in that directory.
haikuporter normally moves the hpkg files from that directory to /path/to/haikuports/packages/ at the end, **_unless_** it gets stuck in the infinite loop.
So if you forget to move manually the hpkg files from that directory to the /path/to/haikuports/packages/ directory and then relaunch haikuporter for that recipe then the hpkgs in work-\*/hpkgs/ are deleted... and you have to cross your fingers again ;-)

**Comment by ohnx** [comment]
Well, apparently capital letters can cause packages to get stuck on activating....

Just renamed my package to use lowercase and it works now.

**Comment by jmairboeck** [comment]
Is this a duplicate of #207 (and thus fixed)?

**Comment by Begasus** [comment]
Same here (even had a few occasional ones on 64bit not that long ago), deffenitly not fixed.
